### PR TITLE
Parse unions without a "type" field

### PR DESCRIPTION
**Pull request by sfackler**
palantir/conjure#111

### CODE REVIEW

File: conjure-test/src/test/objects.rs [hunk 1]
```diff
@@ -128,6 +128,16 @@ fn unions() {
     test_ser(&unknown_value, unknown_json);
 }
 
+#[test]
+fn typeless_unions() {
```
Comment (bavardage) [inline conjure-test/src/test/objects.rs]: worth testing the case of deserializing empty optional? (one of the edge cases described in  https://github.com/palantir/conjure/issues/111)

**Comment by bavardage** [comment]
lgtm, but I guess hold off on merging until https://github.com/palantir/conjure/issues/111 has been resolved definitively
